### ISSUE TITLE
confusing error message for forgetting to wrap a string in quotes

### ISSUE THREAD

**Issue by Gankra** [kdl-org/kdl-rs#58]
given the following:

```rust
struct Complex {

}
```

kdl spits out the following error:

```
Error: 
  × Expected valid node terminator.
   ╭─[7:1]
 7 │
 8 │ struct Complex {
   · ───┬───
   ·    ╰── parsed node
   ╰────
  help: Nodes can only be terminated by `;` or a valid line ending.
  ```
  
It sees `Complex` is a kdl ident and thinks you're trying to start a new node without finishing the `struct` one, but really you just forgot to wrap `Complex` in quotes. I'm not 100% sure about the best way to detect this situation and provide a suggestion. It's not *wrong* and there's some cases where it's definitely the right error.

I think maybe framing it as "Complex is wrong" might work better?

```
Error: 
  × Found a bare identifier in a node's args
   ╭─[7:1]
 7 │
 8 │ struct Complex {
   ·        ───┬───
   ·           ╰── this isn't valid
   ╰────
  help: If this was supposed to be a string, wrap it in quotes. 
        If this was supposed to be a new node, close the previous node with `;` or `}`.
  ```
  
I'll ponder if this kind of detection is easy to add.

**Comment by Gankra** [comment]
For reference I am constantly making this mistake typing out examples for kdlscript because I made it look SO CLOSE to rust that my brain forgets I'm writing kdl lol.

**Comment by zkat** [comment]
oh this one's gonna be tricky to get a good error out of 🤔

**Comment by Gankra** [comment]
I think maybe one good way to detect it is to extend the internal grammar to accept idents in argument position and then emit an error whenever you successfully parse such an item.

But I'm not sure if there's some fundamental ambiguity you're trying to avoid by requiring quoting of strings.

**Comment by zkat** [comment]
No, I think your idea might actually work pretty well! I can't think of anything in the grammar that would prevent idents from being treated as string values.

**Comment by Gankra** [comment]
Nice I got it working:

```
thread 'parser::node_tests::errant_ident' panicked at 'failed to parse:
  × Expected a valid node entry.
   ╭────
 1 │ struct Vec { }
   ·        ─┬─
   ·         ╰── plain identifiers can't be used here
   ╰────
  help: If this was supposed to be a string, wrap it in quotes.
        If this was supposed to be a new node, terminate the previous node with `;` or `}`
 ```

**Comment by zkat** [comment]
This feature has been tagged and released as `kdl@4.6.0` :)
